modules/auxiliary/admin/ldap/rbcd.rb

```diff
@@ -0,0 +1,260 @@
+          ['URL', 'https://www.thehacker.recipes/ad/movement/kerberos/delegations/rbcd'],
+          ['URL', 'https://github.com/SecureAuthCorp/impacket/blob/3c6713e309cae871d685fa443d3e21b7026a2155/examples/rbcd.py']
+        ],
+        'License' => MSF_LICENSE,
+        'Actions' => [
+          ['FLUSH', { 'Description' => 'Delete the security descriptor' }],
+          ['READ', { 'Description' => 'Read the security descriptor' }],
+          ['REMOVE', { 'Description' => 'Remove matching ACEs from the security descriptor DACL' }],
+          ['WRITE', { 'Description' => 'Add an ACE to the security descriptor DACL' }]
+        ],
+        'DefaultAction' => 'READ',
+        'Notes' => {
+          'Stability' => [],
+          'SideEffects' => [CONFIG_CHANGES], # REMOVE, FLUSH, WRITE all make changes
+          'Reliability' => []
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('DELEGATE_TO', [ true, 'The delegation target' ]),
+      OptString.new('DELEGATE_FROM', [ false, 'The delegation source' ])
+    ])
+  end
+
+  def build_ace(sid)
+    Rex::Proto::MsDtyp::MsDtypAccessAllowedAce.new({
+      body: {
+        access_mask: Rex::Proto::MsDtyp::MsDtypAccessMask::ALL,
+        sid: sid
+      }
+    })
+  end
+
+  def fail_with_ldap_error(message)
+    ldap_result = @ldap.as_json['result']['ldap_result']
+    return if ldap_result['resultCode'] == 0
+
+    print_error(message)
+    # Codes taken from https://ldap.com/ldap-result-code-reference-core-ldapv3-result-codes
+    case ldap_result['resultCode']
+    when 1
+      fail_with(Failure::Unknown, "An LDAP operational error occurred. The error was: #{ldap_result['errorMessage'].strip}")
+    when 16
+      fail_with(Failure::NotFound, 'The LDAP operation failed because the referenced attribute does not exist.')
+    when 50
+      fail_with(Failure::NoAccess, 'The LDAP operation failed due to insufficient access rights.')
+    when 51
+      fail_with(Failure::UnexpectedReply, 'The LDAP operation failed because the server is too busy to perform the request.')
+    when 52
+      fail_with(Failure::UnexpectedReply, 'The LDAP operation failed because the server is not currently available to process the request.')
+    when 53
+      fail_with(Failure::UnexpectedReply, 'The LDAP operation failed because the server is unwilling to perform the request.')
+    when 64
+      fail_with(Failure::Unknown, 'The LDAP operation failed due to a naming violation.')
+    when 65
+      fail_with(Failure::Unknown, 'The LDAP operation failed due to an object class violation.')
+    end
+
+    fail_with(Failure::Unknown, "Unknown LDAP error occurred: result: #{ldap_result['resultCode']} message: #{ldap_result['errorMessage'].strip}")
+  end
+
+  def get_delegate_from_obj
+    delegate_from = datastore['DELEGATE_FROM']
+    delegate_from << '$' unless delegate_from.end_with?('$')
```

Comment:
```suggestion
    delegate_from += '$' unless delegate_from.end_with?('$')
```

Just to avoid the mutation here

Reply:
See above. https://github.com/rapid7/metasploit-framework/pull/17181#discussion_r1004500297